DllImportGenerator/DllImportGenerator/Marshalling/StringMarshaller.Utf16.cs

```diff
@@ -0,0 +1,298 @@
+                                                                ArrayRankSpecifier(SingletonSeparatedList(lengthWithNullTerminator)))))))))),
+                                // fixed (char* <fixedCharIdentifier> = <managedIdentifier>)
+                                // {
+                                //     Buffer.MemoryCopy(<fixedCharIdentifier>, <stackAllocIdentifier>, <byteLenIdentifier>, <byteLenIdentifier>);
+                                // }
+                                FixedStatement(
+                                    VariableDeclaration(
+                                        PointerType(PredefinedType(Token(SyntaxKind.CharKeyword))),
+                                        SingletonSeparatedList<VariableDeclaratorSyntax>(
+                                            VariableDeclarator(fixedCharIdentifier)
+                                            .WithInitializer(EqualsValueClause(IdentifierName(managedIdentifier))))),
+                                    Block(
+                                        ExpressionStatement(
+                                            InvocationExpression(
+                                                MemberAccessExpression(
+                                                    SyntaxKind.SimpleMemberAccessExpression,
+                                                    ParseTypeName("System.Buffer"),
+                                                    IdentifierName("MemoryCopy")),
+                                                ArgumentList(
+                                                    SeparatedList(new ArgumentSyntax[] {
+                                                        Argument(IdentifierName(fixedCharIdentifier)),
+                                                        Argument(IdentifierName(stackAllocPtrIdentifier)),
+                                                        Argument(IdentifierName(byteLenIdentifier)),
+                                                        Argument(IdentifierName(byteLenIdentifier))})))))),
+                                // <nativeIdentifier> = <stackAllocIdentifier>;
+                                ExpressionStatement(
+                                    AssignmentExpression(
+                                        SyntaxKind.SimpleAssignmentExpression,
+                                        IdentifierName(nativeIdentifier),
+                                        IdentifierName(stackAllocPtrIdentifier))));
+
+                            // if (<managedIdentifier> == null)
+                            // {
+                            //     <nativeIdentifier> = null;
+                            // }
+                            // else
+                            // {
+                            //     int <byteLenIdentifier> = (<managedIdentifier>.Length + 1) * sizeof(ushort);
+                            //     if (<byteLenIdentifier> > <StackAllocBytesThreshold>)
```

Comment:
Maybe we want to abstract out the conditional stackalloc concept to a base class? We already do this twice and arrays will do this as well.

Reply:
Yep, that is the plan - was going to do it in a subsequent change

Reply:
Ideally, this would just call https://github.com/dotnet/runtime/issues/25423